crates/aptos-crypto/src/multi_ed25519.rs

```diff
@@ -116,6 +117,75 @@ impl MultiEd25519PublicKey {
     pub fn to_bytes(&self) -> Vec<u8> {
         to_bytes(&self.public_keys, self.threshold)
     }
+
+    /// Like `validate_bytes_count_checks`, but only returns the number of sub PKs that passed
+    /// the validation criteria (i.e., deserialized correctly and were not in a small subgroup).
+    pub fn validate_bytes(bytes: &[u8]) -> bool {
+        let mut num_deserializations = 0;
+        let mut num_small_order_checks = 0;
+        MultiEd25519PublicKey::validate_bytes_and_count_checks(
+            bytes,
+            &mut num_deserializations,
+            &mut num_small_order_checks,
+        )
+    }
+
+    /// Checks that all sub PKs would deserialize correctly and are NOT in the small order subgroup.
+    ///
+    /// This function is called by our MultiEd25519 PK validation APIs in Move.
+    ///
+    /// We cannot rely on `TryFrom<&[u8]> for MultiEd25519PublicKey` since it does not exclude
+    /// sub-PKs that are of small order. (Due to limitations in `ed25519_dalek`'s APIs, it cannot
+    /// call the small-subgroup check API on an `ed25519_dalek::PublicKey` struct.) As a result, we
+    /// are forced to replicate some of its code here.
+    ///
+    /// Returns true if the bytes represent a valid MultiEd25519 PK and false otherwise.
+    /// Returns the number of deserialization attempts on a sub PK in `num_deserializations` and the
+    /// number of small order checks on a successfully-deserialized PK in `num_small_order_checks`.
+    ///
+    /// This function returns early as soon as a sub PK fails a check.
+    pub fn validate_bytes_and_count_checks(
+        bytes: &[u8],
+        num_deserializations: &mut usize,
+        num_small_order_checks: &mut usize,
```

Comment:
[low-priority] also make `num_small_order_checks` and `num_small_order_checks` as return values?

So when callers don't care about `num_deser` and `num_small_order`, they can do `(valid, _, _) = validate_bytes_and_count_checks(...)` and save 2 variable definitions.

Reply:
Fixed.